client/src/pages/Signup/components/SignupForm.tsx

```diff
@@ -0,0 +1,279 @@
+import * as React from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useFormik } from 'formik';
+import * as yup from 'yup';
+import { Typography, TextField, Alert, Snackbar } from '@mui/material';
+import axios, { AxiosResponse, AxiosError } from 'axios';
+
+import { PrimaryButton } from '../../../assets/styles/styles';
+import { theme } from '../../../assets/styles';
+import { signup } from '../../../api/UserAPI';
+
+//=============== interfaces ===============//
```

Comment:
nitpick: inconsistent comment style with other components (same for functions and const comment)

i think if we need dividers then there is too much code in one file